lwt/xen_api_lwt_unix.ml

```diff
@@ -62,29 +62,25 @@ module Lwt_unix_IO = struct
     Ssl.create_context Ssl.TLSv1_2 Ssl.Client_context
 
   let open_connection uri =
-    let domain_addr_t = match Uri.host uri with
-      | Some host ->
-        Lwt.catch
-          (fun () ->
-             Lwt_unix.gethostbyname host >>= fun host_entry ->
-             return (host_entry.Lwt_unix.h_addrtype, host_entry.Lwt_unix.h_addr_list.(0)))
-          (fun _ -> fail (Failed_to_resolve_hostname host))
-      | None -> fail (Failed_to_resolve_hostname "") in
     (match Uri.scheme uri with
      | Some "http" -> return false
      | Some "https" -> return true
      | Some "file" -> return false
      | Some x -> fail (Unsupported_scheme x)
      | None -> fail (Unsupported_scheme "")) >>= fun ssl ->
-    let port = match Uri.port uri with
-      | Some x -> x
-      | None -> if ssl then 443 else 80 in
     (match Uri.scheme uri with
      | Some "file" ->
        return (Unix.PF_UNIX, Unix.ADDR_UNIX (Uri.path uri))
      | Some "http" | Some "https" ->
-       domain_addr_t >>= fun (domain, addr) ->
-       return (domain, Unix.ADDR_INET(addr, port))
+      Util.sockaddr_of_uri uri >>= fun sockaddr ->
+        let sockaddr =
+          match sockaddr with
+          | None ->
+            raise Not_found
```

Comment:
Previously the code returned a failed promise when this happened, now an exception is raised. I don't know if this is reasonable or not.

To keep the previous behaviour you can change Utils.sockaddr_of_uri to:
```
  Lwt.catch (fun () ->
    Lwt_unix.getaddrinfo host (string_of_int port) [] >|= function
      | [] ->
        fail (Failed_to_resolve_hostname "")
      | addrinfo :: _ ->
        return addrinfo.Unix.ai_addr)
    (fun _ ->  fail (Failed_to_resolve_hostname host))
```
this simplifies the code in here:
```
      Util.sockaddr_of_uri uri >>= fun sockaddr ->
      return ((Unix.domain_of_sockaddr sockaddr), sockaddr)
```